pkg/apis/admission/types.go

```diff
@@ -19,72 +19,73 @@ package admission
+	// otherwise identical (parallel requests, requests when earlier requests did not modify etc)
+	UID types.UID
 	// Kind is the type of object being manipulated.  For example: Pod
 	Kind metav1.GroupVersionKind
-	// Name is the name of the object as presented in the request.  On a CREATE operation, the client may omit name and
-	// rely on the server to generate the name.  If that is the case, this method will return the empty string.
-	Name string
-	// Namespace is the namespace associated with the request (if any).
-	Namespace string
-	// Object is the object from the incoming request prior to default values being applied
-	Object runtime.Object
-	// OldObject is the existing object. Only populated for UPDATE requests.
-	OldObject runtime.Object
-	// Operation is the operation being performed
-	Operation Operation
 	// Resource is the name of the resource being requested.  This is not the kind.  For example: pods
 	Resource metav1.GroupVersionResource
 	// SubResource is the name of the subresource being requested.  This is a different resource, scoped to the parent
 	// resource, but it may have a different kind. For instance, /pods has the resource "pods" and the kind "Pod", while
 	// /pods/foo/status has the resource "pods", the sub resource "status", and the kind "Pod" (because status operates on
 	// pods). The binding resource for a pod though may be /pods/foo/binding, which has resource "pods", subresource
 	// "binding", and kind "Binding".
+	// +optional
 	SubResource string
+	// Name is the name of the object as presented in the request.  On a CREATE operation, the client may omit name and
+	// rely on the server to generate the name.  If that is the case, this method will return the empty string.
+	// +optional
+	Name string
+	// Namespace is the namespace associated with the request (if any).
+	// +optional
+	Namespace string
+	// Operation is the operation being performed
+	Operation Operation
 	// UserInfo is information about the requesting user
 	UserInfo authentication.UserInfo
-}
-
-// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
-
-// AdmissionResponse describes an admission response.
-type AdmissionResponse struct {
-	metav1.TypeMeta
-
-	// Status is filled in by the webhook and indicates whether the admission
-	// request should be permitted.
+	// Object is the object from the incoming request prior to default values being applied
 	// +optional
-	Status AdmissionReviewStatus
-
-	// The json-patch body. Must be in compliance with RFC 6902.
+	Object runtime.Object
+	// OldObject is the existing object. Only populated for UPDATE requests.
 	// +optional
-	PatchBody []byte
+	OldObject runtime.Object
 }
 
-// AdmissionReviewStatus describes the status of the admission request.
-type AdmissionReviewStatus struct {
+// AdmissionResponse describes an admission response.
+type AdmissionResponse struct {
 	// Allowed indicates whether or not the admission request was permitted.
```

Comment:
Should we add UID here too?

Reply:
I feel that the UID belongs with the request of the metadata about the request (UserInfo, Operation, Namespace etc). If we are not copying them into the response then it doesn't feel right to copy just the UID in. (If we want to abstract them into a type and reflect that in both request/response, I would be fine with that)

Reply:
> If we are not copying them into the response then it doesn't feel right to copy just the UID in.

Not sure I follow... the point of the uid is to identify the unique request, right? If the point of echoing is to bind the response to a particular request, the uid seems like the right thing to copy (it implies the rest of the request fields)

Reply:
So from the discussions I was in, it seems we want things like UserInfo, Operation, Namespace coming back from the Webhook to the KAS (Just like we want the UID coming back). I'm not objecting to putting the UID in the Response field, so much as saying that I think all the fields want round tripping should be dealt with consistently. So we should either leave all of the in the "Request" field for the return trip (so the KAS can confirm the request the webhook operated on) or we should have the webhook copy all of the fields we want in the return trip into the "Response" object. At that point I would suggest we just package them up to make copying them over a but easier and tidier.